tests/unit/lms/services/oauth2_token_test.py

```diff
@@ -103,6 +117,23 @@ def svc(self, pyramid_request, application_instance):
             pyramid_request.lti_user.user_id,
         )
 
+    @pytest.fixture
+    def save_token(self, db_session, svc):
+        def save_token(service=Service.LMS):
+            svc.save(
+                access_token="access_token",
+                refresh_token="refresh_token",
+                expires_in=1234,
+                service=service,
+            )
+            return db_session.query(OAuth2Token).filter_by(service=service).one()
+
+        return save_token
+
+    @pytest.fixture
+    def try_advisory_transaction_lock(self, patch):
+        return patch("lms.services.oauth2_token.try_advisory_transaction_lock")
```

Comment:
Let's just make sure no tests in this file can be accidentally calling through to the real `try_advisory_transaction_lock()` as that can cause tricky problems:

```suggestion
@pytest.fixture(autouse=True)
def try_advisory_transaction_lock(patch):
    return patch("lms.services.oauth2_token.try_advisory_transaction_lock")
```

(And move it to the bottom of the file. This is how we generally do patch fixtures: as `autouse` fixtures at the bottom of the file.)